# Анализ фильмов IMDb 🎬
В этом проекте мы исследуем датасет фильмов IMDb: 
посчитаем основные статистики, посмотрим популярные жанры, 
найдём фильм с максимальной прибылью и построим графики.


In [2]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# Настройка отображения графиков
sns.set(style="darkgrid")


In [4]:
# Загружаем CSV
df = pd.read_csv("IMDB-Movie-Data.csv")

# Первые 5 строк
df.head()


Rank                    Title                     Genre  \
0     1  Guardians of the Galaxy   Action,Adventure,Sci-Fi   
1     2               Prometheus  Adventure,Mystery,Sci-Fi   
2     3                    Split           Horror,Thriller   
3     4                     Sing   Animation,Comedy,Family   
4     5            Suicide Squad  Action,Adventure,Fantasy   

                                         Description              Director  \
0  A group of intergalactic criminals are forced ...            James Gunn   
1  Following clues to the origin of mankind, a te...          Ridley Scott   
2  Three girls are kidnapped by a man with a diag...    M. Night Shyamalan   
3  In a city of humanoid animals, a hustling thea...  Christophe Lourdelet   
4  A secret government agency recruits some of th...            David Ayer   

                                              Actors  Year  Runtime (Minutes)  \
0  Chris Pratt, Vin Diesel, Bradley Cooper, Zoe S...  2014                121   
1  Noomi Rapace, Logan Marshall-Green, Michael Fa...  2012                124   
2  James McAvoy, Anya Taylor-Joy, Haley Lu Richar...  2016                117   
3  Matthew McConaughey,Reese Witherspoon, Seth Ma...  2016                108   
4  Will Smith, Jared Leto, Margot Robbie, Viola D...  2016                123   

   Rating   Votes  Revenue (Millions)  Metascore  
0     8.1  757074              333.13       76.0  
1     7.0  485820              126.46       65.0  
2     7.3  157606              138.12       62.0  
3     7.2   60545              270.32       59.0  
4     6.2  393727              325.02       40.0

## Средний рейтинг фильмов
Посчитаем среднее значение рейтинга по всему датасету.

In [6]:
average_rating = df["Rating"].mean()
print(f"Средний рейтинг фильмов: {average_rating:.2f}")

Средний рейтинг фильмов: 6.72


**Вывод:** Средний рейтинг фильмов составляет примерно 6.72 баллов.

## Сколько фильмов жанра Action?
Учтём, что у фильма может быть несколько жанров.


In [7]:
action_movies = df[df["Genre"].str.contains("Action", na=False)]
print("Фильмов с жанром Action:", len(action_movies))


Фильмов с жанром Action: 303


**Вывод:** В датасете содержится 303 фильмов жанра Action.


## Фильм с максимальной прибылью
Найдём фильм, который принёс наибольший доход.


In [5]:
max_rev_movie = df.loc[df["Revenue (Millions)"].idxmax()]
print("Фильм:", max_rev_movie["Title"])
print("Доход:", max_rev_movie["Revenue (Millions)"], "млн $")


Фильм: Star Wars: Episode VII - The Force Awakens
Доход: 936.63 млн $


**Вывод:** Самым прибыльным фильмом оказался Star Wars: Episode VII - The Force Awakens с доходом 936.63 млн долларов.


## Средняя длительность фильмов


In [8]:
average_runtime = df["Runtime (Minutes)"].mean()
print(f"Средняя длительность фильмов: {average_runtime:.1f} минут")


Средняя длительность фильмов: 113.2 минут


**Вывод:** В среднем фильмы длятся около 113 минут.


## Распределение рейтингов фильмов


In [ ]:
sns.histplot(df["Rating"], bins=20, kde=True)
plt.title("Распределение рейтингов фильмов")
plt.xlabel("Рейтинг")
plt.ylabel("Количество фильмов")
plt.show()
